### PR TITLE
Add Location Service and Tests

### LINKED ISSUE #8: Add Location Service and Tests
# Acceptance Criteria:

- [ ] There is <tt>LocationQueryService.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/services` directory.
- [ ] <tt>LocationQueryService.java</tt>  implements a class that make a call to the appropriate API
      and returns the results.
- [ ] There is  <tt>LocationQueryServiceTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/services` directory.
- [ ] The tests in  <tt>LocationQueryServiceTests.java</tt> provide 100% coverage of  <tt>LocationQueryService.java</tt> for both jacoco and pitest.

# Details

See the assignment description at https://ucsb-cs156.github.io/s24/lab/team01 for details.

### PR DESCRIPTION
**Pull request by emilythai**
In this PR, I added an service that wraps the Location API from

 `https://nominatim.openstreetmap.org/search?q={location}&format=json";`

Closes #8

### CODE REVIEW

File: src/test/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @Autowired
+    private MockRestServiceServer mockRestServiceServer;
+
+    @Autowired
+    private LocationQueryService LocationQueryService;
+
+    @Test
+    public void test_getJSON() {
+
+        String location = "IslaVista";
```
Comment (Jumshim) [inline src/test/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryServiceTests.java:28]: Does this work with a space?

Comment (emilythai) [reply src/test/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryServiceTests.java:28]: It does not work with a space or %20

Comment (Jumshim) [reply src/test/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryServiceTests.java:28]: Alright, sounds good!

**Review by vaaniarora — APPROVED** [review]
lgtm!

**Review by Jumshim — APPROVED** [review]

**Review by rainakakani21 — APPROVED** [review]
lgtm

**Pull request merged**